### ISSUE TITLE
Expected on type argument after "Optional" in "is"-check

### ISSUE THREAD

**Issue by septatrix** [microsoft/pyright#948]
**Describe the bug**
Pyright requires a type argument for generic types even when they are not used for annotations.

**To Reproduce**
Run pyright on the following code
```python
...
if hasattr(type_, "__origin__") and type_.__origin__ is Optional:
    original_type, none_type = type_.__args__
...
```
Output
```
133:65 - error: Expected one type argument after "Optional"
```

**Expected behavior**
This should not generate an error

**VS Code extension or command-line**
Pylance extension and npx to double check (`npx pyright --version` yields `pyright 1.1.63`)

**Additional context**
This syntax should be valid and is used in the official typing docs in the section for e.g. [typing.get_args](https://docs.python.org/3/library/typing.html#typing.get_args)

**Comment by septatrix** [comment]
Okay I think I found the mistake. Optional itself could never be the `__origin__` anyway as it is a shorthand for `Union[X, None]`. Replacing the check with `type_.__origin__ is Union` and afterwards checking if the none type is in the union seems to work
